release/models/rib/openconfig-rib-bgp-tables.yang

```diff
@@ -311,6 +321,639 @@ submodule openconfig-rib-bgp-tables {
+    description "Grouping for type one l2vpn evpn routes";
+    container type-one-ethernet-auto-discovery {
+      description "Top level container for type one l2vpn evpn routes";
+      list type-one-route {
+        key "esi ethernet-tag";
+        description
+          "List of evpn ethernet auto discovery routes";
+
+        uses bgp-evpn-type-one-key-refs;
+
+        container state {
+          description
+            "Operational state data for type1 evpn route entries
+            in the BGP LOC-RIB";
+
+          leaf esi {
+            type oc-evpn-types:esi;
+            description
+              "The ethernet segment identifier for the route";
+          }
+
+          leaf ethernet-tag {
+            type oc-evpn-types:ethernet-tag;
+            description
+              "The ethernet tag for the route";
+          }
+
+          uses bgp-loc-rib-attr-state;
+          uses bgp-common-route-annotations-state;
+          uses bgp-loc-rib-route-annotations-state;
+         }
+
+         uses bgp-evpn-route-path-state;
+         uses bgp-unknown-attr-top;
+      }
+    }
+  }
+
+  grouping bgp-evpn-type-two-key-refs {
+    description
+      "Key references to support operational state structure for
+      BGP EVPN type 2 routes";
+
+      leaf esi {
+        type leafref {
+          path "../state/esi";
+        }
+        description
+          "Reference to the esi list key";
+      }
+
+      leaf ethernet-tag {
+        type leafref {
+          path "../state/ethernet-tag";
+        }
+        description
+          "Reference to the esi list key";
+      }
+
+      leaf mac-address {
+        type leafref {
+          path "../state/mac-address";
+        }
+        description
+          "Reference to the mac-address list key";
+      }
+
+      leaf mac-length {
+        type leafref {
+          path "../state/mac-length";
+        }
+        description
+          "Reference to the mac-length list key";
+      }
+
+      leaf ip-prefix {
+        type leafref {
+          path "../state/ip-prefix";
+        }
+        description
+          "Reference to the ip-prefix list key";
+      }
+
+      leaf ip-length {
+        type leafref {
+          path "../state/ip-length";
+        }
+        description
+          "Reference to the ip-length list key";
+      }
+  }
+
+  grouping bgp-evpn-type-two-state {
+    description "Grouping for type two l2vpn evpn routes";
+    container type-two-mac-ip-advertisement {
+      description "Top level container for type two l2vpn evpn routes";
+      list type-two-route {
+        key "esi ethernet-tag mac-address mac-length ip-prefix ip-length";
+        description
+          "List of evpn mac/ip advertisement routes";
+
+        uses bgp-evpn-type-two-key-refs;
+
+        container state {
+          description
+            "Operational state data for type2 evpn route entries
+            in the BGP LOC-RIB";
+
+          leaf esi {
+            type oc-evpn-types:esi;
```

Comment:
Should this reference the existing ESI?

Reply:
No, this is similar to the RD question.  We need to be able to display local and remote routes.

Reply:
Clarify this in the description